File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/interpreter/Interpreter.java
package interpreter;

import java.util.HashMap;
import java.util.Map;

class Interpreter {
    private Map<String, TerminalExpression> expressions;

    public Interpreter() {
        expressions = new HashMap<>();
        expressions.put("Hello", new TerminalExpression("Bonjour"));
        // Add more expressions as needed
    }

    public void interpret(String expression, Context context) {
        if (expressions.containsKey(expression)) {
            TerminalExpression terminalExpression = expressions.get(expression);
            String translation = terminalExpression.interpret(context);
            System.out.println("Interpreter: Translated expression - " + translation);
        } else {
            System.out.println("Interpreter: Expression not recognized");
        }
    }
}

File: code/csharp/DesignPatterns/interpreter/Interpreter.cs
namespace Interpreter;
using System.Collections.Generic;

class Interpreter
{
    private Dictionary<string, IExpression> _expressions;

    public Interpreter()
    {
        _expressions = new Dictionary<string, IExpression>();
        _expressions.Add("Hello", new TerminalExpression("Bonjour"));
        // Add more expressions as needed
    }

    public void Interpret(string expression, Context context)
    {
        if (_expressions.ContainsKey(expression))
        {
            string translation = _expressions[expression].Interpret(context);
            System.Console
            .WriteLine($"Interpreter: Translated expression - {translation}");
        }
        else
        {
            System.Console.WriteLine("Interpreter: Expression not recognized");
        }
    }
}
